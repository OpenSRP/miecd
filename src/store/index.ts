import { connectReducer, getConnectedStore } from '@onaio/connected-reducer-registry';
import { gateKeeperReducer, gateKeeperReducerName } from '@onaio/gatekeeper';
import reducerRegistry, { combine } from '@onaio/redux-reducer-registry';
import session, { reducerName as sessionReducer } from '@onaio/session-reducer';
import { Dictionary } from '@onaio/utils';

/** Initial reducers in the reducer registry */
const defaultReducers: Dictionary = {
    router: connectReducer /** Dirty hack because connectRouter LocationChangeAction does not extend Redux.AnyAction */,
};

/** Add users reducer to registry */
defaultReducers[sessionReducer] = session;
defaultReducers[gateKeeperReducerName] = gateKeeperReducer;

/** The initial store for the reveal web app */
const store = getConnectedStore(defaultReducers);

/** Set listener to add reducers to store when registered */
reducerRegistry.setChangeListener((reducers) => {
    store.replaceReducer(combine(reducers));
});

export default store;
