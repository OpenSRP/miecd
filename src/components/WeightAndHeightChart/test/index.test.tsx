/* eslint-disable @typescript-eslint/no-explicit-any */
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { act } from 'react-dom/test-utils';
import { Chart } from '..';
import store from '../../../store/index';
import * as Highcharts from 'highcharts';
import { chartArgument } from './fixtures';
const history = createBrowserHistory();

const weights = {
    categories: ['20/2019', '8/2019', '28/2019'],
    dataSeries: [{ data: [10, 10, 9], name: 'Weights' }],
};

const defaultProps = {
    chartWrapperId: 'wrapper-id',
    legendString: 'legend string',
    title: 'the title',
    units: 'kgs',
    xAxisLabel: 'x axis label',
};

describe('Chart', () => {
    beforeEach(() => {
        jest.resetAllMocks();
    });
    // eslint-disable-next-line jest/expect-expect
    it('must render without crashing', () => {
        shallow(<Chart dataArray={weights} {...defaultProps} />);
    });

    it('must render correctly', () => {
        const wrapper = mount(
            <Provider store={store}>
                <Router history={history}>
                    <Chart dataArray={weights} {...defaultProps} />
                </Router>
            </Provider>,
        );
        expect(toJson(wrapper.find('#wrapper-id'))).toMatchSnapshot();
        wrapper.unmount();
    });

    it('calls hicharts.chart with the correct arguments', async () => {
        const div = document.createElement('div');
        document.body.appendChild(div);
        // eslint-disable-next-line @typescript-eslint/ban-types
        const customTimeout = (fun: Function) => {
            fun();
        };
        const actualTimeout = window.setTimeout;
        (window as any).setTimeout = customTimeout;

        const chartMock = jest.fn();
        (Highcharts as any).chart = chartMock;

        const wrapper = mount(
            <Provider store={store}>
                <Router history={history}>
                    <Chart dataArray={weights} {...defaultProps} />
                </Router>
            </Provider>,
            { attachTo: div },
        );

        await act(async () => {
            await new Promise((resolve) => setImmediate(resolve));
        });

        expect(JSON.stringify(chartMock.mock.calls)).toEqual(chartArgument);

        (window as any).setTimeout = actualTimeout;
        wrapper.unmount();
    });
});
