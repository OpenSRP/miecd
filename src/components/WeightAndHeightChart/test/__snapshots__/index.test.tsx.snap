// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart must render correctly 1`] = `
<div
  id="wrapper-id"
/>
`;
